dll/win32/browseui/explorerband.cpp

```diff
@@ -374,28 +374,37 @@ void CExplorerBand::OnSelectionChanged(LPNMTREEVIEW pnmtv)
 
 void CExplorerBand::OnTreeItemDragging(LPNMTREEVIEW pnmtv, BOOL isRightClick)
 {
-    CComPtr<IShellFolder>               pSrcFolder;
-    CComPtr<IDataObject>                pObj;
-    LPCITEMIDLIST                       pLast;
-    HRESULT                             hr;
-    DWORD                               dwEffect;
-    DWORD                               dwEffect2;
-
-    dwEffect = DROPEFFECT_COPY | DROPEFFECT_MOVE;
     if (!pnmtv->itemNew.lParam)
         return;
+
     NodeInfo* pNodeInfo = GetNodeInfo(pnmtv->itemNew.hItem);
+
+    HRESULT hr;
+    CComPtr<IShellFolder> pSrcFolder;
+    LPCITEMIDLIST pLast;
     hr = SHBindToParent(pNodeInfo->absolutePidl, IID_PPV_ARG(IShellFolder, &pSrcFolder), &pLast);
     if (!SUCCEEDED(hr))
         return;
-    hr = pSrcFolder->GetUIObjectOf(m_hWnd, 1, &pLast, IID_IDataObject, 0, reinterpret_cast<void**>(&pObj));
+
+    SFGAOF attrs = SFGAO_CANCOPY | SFGAO_CANMOVE | SFGAO_CANLINK;
+    pSrcFolder->GetAttributesOf(1, &pLast, &attrs);
+
+    DWORD dwEffect = 0, dwEffect2;
```

Comment:
dwEffect2 might be moved to the DoDragDrop call.
Alternatively, I guess you could do:
`DoDragDrop(..., dwEffect, &dwEffect);` to avoid using a second variable.

Reply:
Fixed.